### PR TITLE
Introduce legal name for AccountTransactions

### PR DESCRIPTION
**Pull request by Betree**
Require https://github.com/opencollective/opencollective-api/pull/6612
Resolve https://github.com/opencollective/opencollective/issues/4710

```csv
"datetime","shortGroup","type","kind","isRefund","isRefunded","displayAmount","amount","paymentProcessorFee","netAmount","currency","accountSlug","accountName","oppositeAccountSlug","oppositeAccountName"
"2021-08-20T10:25:02","52b03850","DEBIT","CONTRIBUTION","REFUND","","-$17.77 USD",-17.77,0,-16.88,"USD","apex","APEX","betree","Benjamin Piouffle (Betree)"
"2021-08-20T10:25:02","bf9d9806","CREDIT","HOST_FEE","REFUND","","$1.00 USD",1,0,1,"USD","apex","APEX","opensourceorg","Open Source Collective 501(c)(6) (OSC)"
```

### CODE REVIEW

File: src/server/controllers/account-transactions.js [hunk 1]
```diff
@@ -181,6 +184,18 @@ const formatAmountAsString = (amount) => {
...
+  const name = account?.name;
+  if (!legalName && !name) {
+    return '';
+  } else if (legalName && name) {
+    return `${legalName} (${name})`;
+  } else {
+    return legalName || name;
+  }
+};
+
```
Comment (znarf) [inline src/server/controllers/account-transactions.js:198]: Can we maybe handle the case where legalName === name and skip name in this case?

Already 86 entries there:

```
SELECT * FROM "Collectives" WHERE "legalName" = "name";
```

Comment (Betree) [reply src/server/controllers/account-transactions.js]: Good catch, I changed that. I'll bring this change in frontend and PDF service as well

File: src/server/controllers/account-transactions.js [hunk 2]
```diff
@@ -254,7 +269,9 @@ const defaultFields = [
   'netAmount',
   'currency',
   'accountSlug',
+  'accountName',
   'oppositeAccountSlug',
+  'oppositeAccountName',
```
Comment (znarf) [inline src/server/controllers/account-transactions.js:274]: I'm wondering, should we keep slug if we decide to have name?

Comment (znarf) [reply src/server/controllers/account-transactions.js:274]: Not a blocker but something to have in mind.

**Comment by Gruppresarial** [comment]
Me puede ayudar

File: src/server/controllers/account-transactions.js (hunk 2 above)
Comment (Betree) [reply src/server/controllers/account-transactions.js:274]: @znarf `slug` is still interesting to quickly go from the CSV to the profile on Open Collective. Maybe we could replace `accountSlug` by `accountURL` and make its value `https://opencollective.com/${slug}`?

**Pull request merged**